### PR TITLE
Media Library: Display File Size for Individual Items

### LINKED ISSUE #49492: Nav Unification: Add Media File Size to Media Library
**What**
The Media Library accessed from the Media section of the new Nav Unification dashboard does not show the file size for the individual media items.  I currently refer users to the Media Library on the WP Admin dashboard to see that information. Now that the WP Admin link is gone, it will be harder for users to see how big the individual files in their media library are. 

**Why**
We often recommend that users review the size of their media files and optimize larger images when they complain of slow sites, but it is hard for them to do that when the information isn't readily available. 

**How**
Add the file size somewhere in the area of the label that displays the dimensions.
screenshot
Image URL: https://d.pr/i/zLdR9R

### PR DESCRIPTION
**Pull request by Aurorum**
#### Changes proposed in this Pull Request

* Allows users to see the size of each file they upload

#### Testing instructions

You'll need to apply Automattic/jetpack#18637 for this to work!

Visit the Media Library and open a file, then verify that the "File Size" is displayed neatly in the corner which accurately corresponds to the size stated in WP-Admin. 

Screenshot 2021-01-30 at 20 51 09

Fixes #49492

cc @sgomes

### CODE REVIEW

File: client/post-editor/media-modal/detail/detail-file-info.jsx [hunk 1]
```diff
@@ -102,6 +102,10 @@ class EditorMediaModalDetailFileInfo extends React.Component {
 						<th>{ this.props.translate( 'File Type' ) }</th>
 						<td>{ this.getItemValue( 'extension' ) }</td>
 					</tr>
+					<tr>
+						<th>{ this.props.translate( 'File Size' ) }</th>
```
Comment (a8ci18n) [inline client/post-editor/media-modal/detail/detail-file-info.jsx]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **141** times:
`translate( 'File size:' )` *ES Score: 8*
See 1 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/pull/49500)

**Review by jeherve — CHANGES_REQUESTED** [review]
You'll probably need to avoid displaying the field when size is set to 0, like when that data is missing (old versions of Jetpack, images missing some metadata for some reason).

**Comment by Aurorum** [comment]
Fixed, thank you! :)

File: client/post-editor/media-modal/detail/detail-file-info.jsx [hunk 2]
```diff
@@ -84,6 +84,21 @@ class EditorMediaModalDetailFileInfo extends React.Component {
...
+	renderFileSize = () => {
+		const fileSize = this.getItemValue( 'size' );
+
+		if ( ! fileSize || fileSize === 0 ) {
+			return;
+		}
+
+		return (
+			<tr>
+				<th>{ this.props.translate( 'File Size' ) }</th>
```
Comment (a8ci18n) [inline client/post-editor/media-modal/detail/detail-file-info.jsx:97]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **141** times:
`translate( 'File size:' )` *ES Score: 8*
See 1 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/pull/49500)

**Comment by obenland** [comment]
@jeherve Would you be available for another review on this one?

**Review by jeherve — APPROVED** [review]
This tests well for me now. 👍

**Comment by tjcafferkey** [comment]
Hey @Aurorum thank you so much for wrangling this. It appears since the release of your other PR (thank you for that too) there are some conflicts that need to be resolved.

Is this something you have the time to do? Please let me know if not.

**Comment by Aurorum** [comment]
Hi, those conflicts should be resolved now!

**Review by jeherve — APPROVED** [review]

**Comment by tjcafferkey** [comment]
@Aurorum would you be able to update this branch with the latest changes from `trunk` please? 

There's some changes to the tests which are needed before we can merge this change.

Thank you!

**Comment by Aurorum** [comment]
Sure thing, done! :)

**Review by jeherve — APPROVED** [review]

**Comment by BogdanUngureanu** [comment]
Hey @Aurorum, thank you for your contribution! I've just merged your changes. :)

**Pull request merged**